sd_dynamic_prompts/dynamic_prompting.py

```diff
@@ -417,16 +407,119 @@ def process(
         original_seed = p.seed
         num_images = p.n_iter * p.batch_size
 
+        combinatorial_batches = int(combinatorial_batches)
+        if self._auto_purge_cache:
+            self._wildcard_manager.clear_cache()
+
+        all_prompts, all_negative_prompts = self.generate_prompts(
+            p,
+            original_prompt,
+            original_negative_prompt,
+            original_seed,
+            num_images,
+            is_combinatorial,
+            combinatorial_batches,
+            is_magic_prompt,
+            is_feeling_lucky,
+            is_attention_grabber,
+            min_attention,
+            max_attention,
+            magic_prompt_length,
+            magic_temp_value,
+            use_fixed_seed,
+            unlink_seed_from_prompt,
+            disable_negative_prompt,
+            enable_jinja_templates,
+            max_generations,
+            magic_model,
+            magic_blocklist_regex,
+        )
+
+        updated_count = len(all_prompts)
+        p.n_iter = math.ceil(updated_count / p.batch_size)
+
+        if num_images != updated_count:
+            p.all_seeds, p.all_subseeds = get_seeds(
+                p,
+                updated_count,
+                use_fixed_seed,
+                is_combinatorial,
+                combinatorial_batches,
+            )
+
+        if updated_count > 1:
+            logger.info(
+                f"Prompt matrix will create {updated_count} images in a total of {p.n_iter} batches.",
+            )
+
+        self._prompt_writer.set_data(
+            positive_template=original_prompt,
+            negative_template=original_negative_prompt,
+            positive_prompts=all_prompts,
+            negative_prompts=all_negative_prompts,
+        )
+
+        p.all_prompts = all_prompts
+        p.all_negative_prompts = all_negative_prompts
+        if no_image_generation:
+            logger.debug("No image generation requested - exiting")
+            # Need a minimum of batch size images to avoid errors
+            p.batch_size = 1
+            p.all_prompts = all_prompts[0:1]
+
+        p.prompt_for_display = original_prompt
+        p.prompt = original_prompt
+
+        if hr_fix_enabled:
+            p.all_hr_prompts = _get_hr_fix_prompts(
+                all_prompts,
+                original_hr_prompt,
+                original_prompt,
+            )
+            p.all_hr_negative_prompts = _get_hr_fix_prompts(
+                all_negative_prompts,
+                original_negative_hr_prompt,
+                original_negative_prompt,
+            )
+
+    def generate_prompts(
```

Comment:
Just generating prompts shouldn't require the full `p` object, I don't think..?

Further, maybe this should be made a free function, and it should probably be kwarg-only (`(*, original_prompt...)`) because otherwise it's super easy to pass arguments in the wrong order.

Reply:
`p` is used for its additional seed settings. see around `dynamic_prompting.py.py#558` and aound `dynamic_prompting.py.py#572`.

Reply:
Sure, but those probably ought to be moved on from this function?

Reply:
those are used for the `generate_prompts_helper()`. I'm not seeing an easy way to move them out, any suggestions?

Reply:
Well, `GeneratorBuilder().set_context(p)` is a bit problematic, but that's only used for Jinja context – the p-to-context code in `create_jinja_generator` should be refactored out. (I think Jinja over the API would fail right now since `p.sd_model` will not get set.)

With that done, I the `p.all_seeds, p.all_subseeds = get_seeds(...)` manipulation could be done outside the prompt-generation function, couldn't it?